it/services/ArrivalNotificationServiceSpec.scala

```diff
@@ -0,0 +1,138 @@
+package it.services
+
+import generators.MessageGenerators
+import models.messages.{ArrivalNotification, NormalNotification, SimplifiedNotification}
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.{FreeSpec, MustMatchers, OptionValues}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import play.api.libs.json.{JsObject, Json}
+import reactivemongo.play.json.ImplicitBSONHandlers.JsObjectDocumentWriter
+import reactivemongo.play.json.collection.JSONCollection
+import repositories.CollectionNames
+import services.ArrivalNotificationService
+import services.mocks.MockDateTimeService
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class ArrivalNotificationServiceSpec
+  extends FreeSpec
+    with MustMatchers
+    with MongoSuite
+    with ScalaFutures
+    with GuiceOneAppPerSuite
+    with IntegrationPatience
+    with MockDateTimeService
+    with OptionValues
+    with ScalaCheckPropertyChecks
+    with MessageGenerators {
+
+  val service: ArrivalNotificationService = app.injector.instanceOf[ArrivalNotificationService]
+
+  "ArrivalNotificationService" - {
+
+    "must persist NormalNotification within mongoDB" in {
+
+      forAll(arbitrary[NormalNotification]) {
+
+        normalNotification =>
+
+          database.flatMap(_.drop()).futureValue
```

Comment:
I know we use this in a couple of places. Could we mix in BeforeAndAfter trait and use the beforeEach method to drop the database.